### DISCUSSION TITLE
pre-release tag on the installed extension details page in vscode

### DISCUSSION THREAD

**Discussion by cgrama** [microsoft/vscode-discussions#1221] [Extension Development QnA] [answered]
Before publishing a pre-release extension to marketplace, how can we verify if the extension has been packaged as a pre-release extension by "vsce package --pre-release" ?
We have done the following checks to verify if the extension has been packaged as a pre-release:
1. Installed the pre-release extension package in vscode
2. Verify  the <extensions installed folder>\<extension name\.vsixmanifest file ha the following prerelease property
       <Property Id="Microsoft.VisualStudio.Code.PreRelease" Value="true" />
3. In the vscode extensions view, when we select this installed pre-release extension:
    a. In the extensions view, the pre-release image is not shown to the left of the extension 
    b. In the extensions view, there is no "Install Pre-release version" action for the extension
    c. the extension's home/landing page in vscode does not show the "pre-release" label/tag
    d. the extension's home/landing page  in vscode does not show "Install Release version / Install Pre-Release version" actions
When will the info in 2.a to 2.d be shown for a pre-release extension ?
Will this info be shown only after the pre-release extension is published to the marketplace ?
What additional verification can extension developers do to make sure the extension is pre-release before "vsce publish --pre-release" of the extension ?

**Comment by tamuratak** [comment]
The generated `.vsix` file is a zip file. You can verify the content with unzip if you want.

**Reply by isidorn** [reply]
@cgrama I will go ahead and mark this as answer. In case this does not work for you. let us know

**Reply by cgrama** [reply]
To summarize:
To prepare a vscode extension for the FIRST pre-release version to be published to the marketplace:
- package the extension with "vsce package --prerelease"
- check the vsixmanifest file in the vsix generated from the above vsce command
- there is no other way to verify the pre-release version of the extension before publishing the pre-release to the vscode extension

**Reply by isidorn** [reply, marked as answer]
You can also just drag and drop your packaged vsix into VS Code (that does not have your extension installed). You will then see how VS Code renders your extension version, and if it is pre-release.
